okhttp/src/main/java/okhttp3/internal/ws/RealWebSocket.kt

```diff
@@ -147,6 +162,11 @@ class RealWebSocket(
         .header("Connection", "Upgrade")
         .header("Sec-WebSocket-Key", key)
         .header("Sec-WebSocket-Version", "13")
+        .apply {
+          if (originalRequest.header(HEADER_WS_EXTENSION) == null) {
+            header(HEADER_WS_EXTENSION, HEADER_VALUE_CONTEXT_TAKEOVER)
```

Comment:
I think we need an else clause here. If the extension is already present, we need to do one of two things:

 - disable our code to handle compression
 - fail outright

My preference is to fail outright; perhaps very early. Unlike HTTP compression, there’s no way the application layer can effectively implement WebSocket extensions on top of OkHttp’s APIs. This is because they don’t get access to the reserved bits.

So - - - I propose we change OkHttp to explode if the WebSocket Request has a `HEADER_WS_EXTENSION` set.